### PR TITLE
Show missing Hour of Code events on hourofcode.com/us

### PR DESCRIPTION
**Pull request by TurnerRiley**
Several Hour of Code events were not showing up on the [Hour of Code map](https://hourofcode.com/us) despite having location data associated with them. For example, the event for [this Zendesk ticket](https://codeorg.zendesk.com/agent/tickets/470749) is not on the map. I tracked down the issue to the Hour of Code event form's "Event location" field with the location address dropdown. While the dropdown provides processed location data (such as the long/lat coordinates) for the user's selected address, it does not always fill in every field. In particular, it does not always fill in the `location_city_s` field.

This field is currently required for a regular event (i.e. not a special event that requires approval) to be placed on the map, otherwise it is skipped. I locally added a form entry with the exact same fields as the one in the Zendesk to confirm that it was being skipped at that step and not placed on the map.

I checked with Dani for a product perspective on the desired outcome, and she said we should just remove the line that skips those events. So, this PR removes the line which skips regular events with a nil `location_city_s` field. This is safe to do as the only place this field is being used is to populate what's in paranthesis when clicking on an event on the Hour of Code map (e.g. Bayard in "COBRE HIGH (Bayard)"). I thought this would also require updating [hoc_events_map.js](https://github.com/code-dot-org/code-dot-org/blob/staging-next/apps/src/sites/hourofcode.com/pages/views/hoc_events_map.js) (where the map layers are rendered) to support the case where a nil `location_city_s` field is passed in, however the file [already supports it](https://github.com/code-dot-org/code-dot-org/blob/staging-next/apps/src/sites/hourofcode.com/pages/views/hoc_events_map.js#L38) by not printing "(city_name)" if the city is nil.

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-1167)
Zendesk example: [here](https://codeorg.zendesk.com/agent/tickets/470749)

## Testing story
I attempted to test this similar to the way I tested the [`update_census_mapbox` cronjob PR](https://github.com/code-dot-org/code-dot-org/pull/53972) (i.e. setting up an adhoc to run the cronjob on a developer set of mapbox tiles, then locally running hourofcode.com and pointing my local mapbox to the developer set of mapbo tiles that were updated (this setup is required because I cannot locally run the mapbox cronjob, and adhocs do not support hourofcode.com)). Unfortunately, I could not get the `update_hoc_map` cronjob to successfully run on adhoc due to mapbox errors.

As a small proof of concept, I added `puts` statements within the cronjob (but before the upload to mapbox where the errors were occurring) to check whether the event was being skipped (as it was before) or if it was being processed (desired outcome). I ran it and confirmed that the new logic allowed the event to be processed and "Not skipped" was successfully printed in the cronjob output:
not_skipped

While this does give me confidence that events missing a `location_city_s` field will no longer be skipped, it does not explicitly show that it will then make it onto the map without an error. Since the skipping was the reason it wasn't making it onto the map in the first place and since the `hoc_events_map.js` already has coverage for a nil city field, I believe it is still a safe change to make (but I am fully open to pushback or other testing requests if others have ideas)!

### CODE REVIEW

File: bin/cron/update_hoc_map [hunk 1]
```diff
@@ -35,11 +35,9 @@ def main
     # update our JS code (hoc_events_map.js) that visualizes these events
     # to handle missing data.
     # ***
-    # In order to be placed on the map, all events require an organization name,
-    # processed location, and city. With the exception that approved special events
-    # don't need a city so we can show events like the Cocos Keeling Islands.
+    # In order to be placed on the map, all events require an organization name and
+    # a processed location.
     next if data['organization_name_s'].nil? || processed_data['location_p'].nil?
-    next if form[:review] != "approved" && processed_data['location_city_s'].nil?
```
Comment (bethanyaconnor) [inline bin/cron/update_hoc_map:42]: Do we still want to skip it if the event isn't approved? (I don't know how approval works...)

**Review by bethanyaconnor — COMMENTED** [review]
One question, but your testing strategy seems reasonable to me!

**Review by bethanyaconnor — APPROVED** [review]

File: bin/cron/update_hoc_map (hunk 1 above)
Comment (bethanyaconnor) [reply bin/cron/update_hoc_map:42]: Closing this look: I learned that `approved` is what indicates that this is a special event. So, this does what the PR description indicates -- removes the requirement for a city from non-special events

**Review by cat5inthecradle — APPROVED** [review]

**Review by jmkulwik — APPROVED** [review]

**Comment by breville** [comment]
Thanks for tackling this legacy code!

**Pull request merged**